File: backend/models/Insurance.js
const mongoose = require('mongoose');

const insuranceSchema = new mongoose.Schema({
  car: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Car',
    required: true
  },
  owner: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  policyNumber: {
    type: String,
    required: true,
    unique: true
  },
  provider: {
    type: String,
    required: true
  },
  coverageType: {
    type: String,
    required: true,
    enum: ['Comprehensive', 'Third Party', 'Zero Depreciation']
  },
  startDate: {
    type: Date,
    required: true
  },
  endDate: {
    type: Date,
    required: true
  },
  premium: {
    type: Number,
    required: true
  },
  coverageAmount: {
    type: Number,
    required: true
  },
  documents: [{
    type: String
  }],
  isActive: {
    type: Boolean,
    default: true
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('Insurance', insuranceSchema);

File: backend-java/src/main/java/com/carrental/model/Insurance.java
package com.carrental.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@Document(collection = "insurances")
public class Insurance {

    @Id
    private String id;
    
    private String carId;
    
    private String ownerId;
    
    private String policyNumber;
    
    private String provider;
    
    private String coverageType; // Comprehensive, Third Party, Zero Depreciation
    
    private Date startDate;
    
    private Date endDate;
    
    private Double premium;
    
    private Double coverageAmount;
    
    private List<String> documents;
    
    private boolean isActive = true;
    
    private Date createdAt = new Date();
}
